src/components/Documents.tsx

```diff
@@ -1,6 +1,33 @@
-import { Grid, GridItem, Image } from "@chakra-ui/react";
+import {
+  Grid,
+  GridItem,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  Button,
+  Input,
+  Image,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+
+import DocText from "./DocText";
 
 function Documents() {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const list: React.ReactElement[] = [];
+  const [docs, setDocs] = useState(list);
```

Comment:
This is an interesting way of keeping state (storing components themselves) - usually it is better to store the data that the component relies on itself. So storing an array/list of document data JSONs would be a better practice in this case, as you are able to use that data for other needs in the code + have a better structure for state.